src/squlearn/encoding_circuit/qcnn_encoding_circuit.py

```diff
@@ -0,0 +1,686 @@
+        default (bool): If True, the default circuit is built.
+    """
+
+    def __init__(self, num_qubits: int = 0, num_features: int = 0, default: bool = False) -> None:
+        super().__init__(num_qubits, num_features)
+        self._num_parameters = 0
+        self._left_qubits = [i for i in range(num_qubits)]
+        self._operations_list = []
+        self._default = default
+        if default:
+            if num_qubits == 0:
+                print("To generate a default circuit provide a number of qubits > 0.")
+            else:
+                self.default_circuit()
+
+    @property
+    def num_parameters(self) -> int:
+        """Returns the number of trainable parameters of the current encoding circuit."""
+        return self._num_parameters
+
+    @property
+    def left_qubits(self) -> list:
+        """Returns the qubits which one can operate on in the current circuit."""
+        return self._left_qubits
+
+    @property
+    def operations_list(self) -> list:
+        """Returns the list of operators currently acting on the encoding circuit."""
+        return self._operations_list
+
+    def set_num_qubits(self, n: int = 0):
+        """
+        Set the number of qubits of the circuit.
+
+        If the number of qubits is reduced and the supplied pooling gates do not fit to this,
+        this will throw a error in the troubling layer.
+
+        Args:
+            n (int): The number of qubits in the new circuit.
+        """
+
+        if n > 0:
+            if self.num_qubits == n:
+                print("The number of qubits did not change.")
+            else:
+                self._num_qubits = n
+                self._left_qubits = [i for i in range(n)]
+                self._num_parameters = 0
+                if self._default:
+                    self._operations_list = []
+                    self.default_circuit()
+                else:
+                    for operation in self.operations_list:
+                        if operation[0] == "Conv":
+                            self.convolution(*operation[1:], _new_operation=False)
+                        elif operation[0] == "Pool":
+                            self.pooling(*operation[1:], _new_operation=False)
+                        elif operation[0] == "FC":
+                            self.fully_connected(*operation[1:], _new_operation=False)
+                            break  # since FC should be at the end of the circuit
+
+    def set_num_features(self, n: int = 0):
+        """
+        Change the number of features in the feature map of the circuit.
+
+        Args:
+            n (int): The new number of features.
+        """
+
+        if n > 0:
+            self._num_features = n
+
+    def convolution(
+        self,
+        quantum_circuit: Union[QuantumCircuit, EncodingCircuitBase, None] = None,
+        label: str = "Conv",
+        alternating: bool = True,
+        diff_params: bool = True,
+        _new_operation: bool = True,
```

Comment:
Docstring description for `_new_operation` is missing. This also applies for other functions.